### PR TITLE
[stable/spinnaker] Option to use external redis

### LINKED ISSUE #9777: spinnaker: allow for external redis?
**Is this a request for help?**: yes

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): feature-request

**Version of Helm and Kubernetes**: `2.10.x`, `1.10.x`

**Which chart**: spinnaker

**What happened**: couldn't specify external redis

**What you expected to happen**: can specify external redis

**How to reproduce it** (as minimally and precisely as possible): attempt to specify external redis

**Anything else we need to know**:

hi guys, (@dwardu89 , @viglesiasce , @lwander)

just kicking the tires on spinnaker and associated helm chart.

reading [this](https://www.spinnaker.io/setup/productionize/caching/externalize-redis/), i was thinking it might be a helpful option to support an externalized redis (e.g. [elasticache](https://aws.amazon.com/elasticache/)) via the helm chart.

i see the current helm chart lists redis in `requirements.yaml` and also hardcodes a reference to that instance [here](https://github.com/helm/charts/blob/master/stable/spinnaker/templates/statefulsets/halyard.yaml#L40).

i'm kind of an intermediate helm guy, i can imagine how [this reference](https://github.com/helm/charts/blob/master/stable/spinnaker/templates/statefulsets/halyard.yaml#L40) to a redis instance can be made configurable by the chart user, but not sure if something in `requirements.yaml` can be made optional?

i would be happy to attempt a p/r if you guys thought it might be a beneficial option, but wanted to reach out for any thoughts by peeps with more experience before moving forward.

best,
tony

### PR DESCRIPTION
**Pull request by vyasswaroop**
Signed-off-by: Vyas Swaroop

#### What this PR does / why we need it: 
Adds an option for Spinnaker to use external Redis cluster. In spinnaker documentation itself, it is recommended to use an external Redis cluster for higher reliability to make it production-ready.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #9777

#### Special notes for your reviewer:
Created extra parameter in spinnaker values file in `redis` section to enable use of an external redis cluster as well as parameters to specify the host and port of the Redis cluster.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by rimusz** [comment]
also state in the issue subject which charts PR belongs to: as `[stable/spinnaker] Option to use external redis`

File: stable/spinnaker/values.yaml [hunk 1]
```diff
@@ -104,6 +104,11 @@ redis:
   nodeSelector: {}
   cluster:
     enabled: false
+# Change this to true if youd like to use external redis
+  external: false
```
Comment (paulczar) [inline stable/spinnaker/values.yaml]: usually when we add support for external alternatives we would have `redis.enabled: true` and then we'd update the requirements.yaml to determine if it should actually include the redis chart ( see the minio conditional right below it https://github.com/helm/charts/blob/master/stable/spinnaker/requirements.yaml#L8 ).

I'd like to see you use that same pattern.

**Review by paulczar — CHANGES_REQUESTED** [review]

File: stable/spinnaker/values.yaml (hunk 1 above)
Comment (vyasswaroop) [reply stable/spinnaker/values.yaml]: Yup. Thanks for the change request. I'll update the redis value structure similar to the one used in minio, and will also update the requirements file.

File: stable/spinnaker/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: Open source, multi-cloud continuous delivery platform for releasing software changes with high velocity and confidence.
 name: spinnaker
-version: 1.1.7
+version: 1.1.8
```
Comment (cpanato) [inline stable/spinnaker/Chart.yaml]: adding a new functionality 
```suggestion
version: 1.2.0
```

Comment (vyasswaroop) [reply stable/spinnaker/Chart.yaml]: Thanks. I've changed that to 1.2.1.

**Comment by vyasswaroop** [comment]
@cpanato I've signed the DCO. Currently the CI test is failing and showing the error as Error: requirements.lock is out of sync with requirements.yaml. Could you please help.

**Review by paulczar — APPROVED** [review]

File: stable/spinnaker/values.yaml [hunk 3]
```diff
@@ -122,8 +122,14 @@ spinnakerFeatureFlags:
 nodeSelector: {}
 
 # Redis password to use for the in-cluster redis service
-# Redis is not exposed publically
+# Enable redis to use in-cluster redis
 redis:
+  enabled: true
+  # External Redis option will be enabled if in-cluster redis is disabled
+  external:
+    host: "<EXTERNAL-REDIS-HOST-NAME>"
```
Comment (cpanato) [inline stable/spinnaker/values.yaml:130]: maybe can comment this out to avoid confusion

**Comment by paulczar** [comment]
I've rebased this on top of master and fixed the versioning.  should be good to merge now!

/lgtm

**Pull request merged**